### PR TITLE
sci-physics/lhapdf: Patch CXXFLAGS and LDFLAGS

### PR DESCRIPTION
**Pull request by APN-Pucky**
Bug: https://bugs.gentoo.org/875386
Bug: https://bugs.gentoo.org/873655
Bug: https://bugs.gentoo.org/873652
Signed-off-by: Alexander Puck Neuwirth <alexander@neuwirth-informatik.de>

This addresses QA warnings of CFLAGS and LDFLAGS not being set for the python binding lib lhapdf.so:

I also created a PR to upstream: https://gitlab.com/hepcedar/lhapdf/-/merge_requests/39

Is there something like a post install test for ebuilds?

I tested manually that the python-install-command works well:
```sh
$ lhapdf install cteq66
```
and that the python library does not just return rubbish:
```py
>>> import lhapdf
>>> print(lhapdf.__version__)
6.5.3
>>> pdfset = lhapdf.getPDFSet('cteq66')
>>> u = pdfset.uncertainty([1 for k in range(pdfset.size)])
>>> print(u.central)
1.0
```

### CODE REVIEW

**Comment by APN-Pucky** [comment]
Oh, I also bumped to 6.5.3

**Comment by thesamesam** [comment]
Thanks. Would you mind either splitting it into two commits (and revbumping the old versions, `git mv` may be fine here given it's clearly wrong before) or just only touching the new version (and adjusting the commit message accordingly)?

My preference would be to git mv all old ones & fix them in one commit, then another new commit to just bump.

**Comment by thesamesam** [comment]
>Is there something like a post install test for ebuilds?

No, we only have src_test, which runs before src_install. Could you possibly write and run a tiny Python program in there? That could work.

**Comment by APN-Pucky** [comment]
> No, we only have src_test, which runs before src_install (see https://bugs.gentoo.org/402581). Could you possibly write and run a tiny Python program in there? That could work.

Thanks. I don't think that helps here, since I also wanted to check that their tools installs the py-so-lib into a correct directory.
Some previous version installed the `*.so` into `$prefix/lib/python3.X/site-packages/LHAPDF-$LHAPDF_VERSION-py3.X-linux-x86_64.egg/` and now `$prefix/lib/python3.X/site-packages/`.

**Comment by thesamesam** [comment]
> > No, we only have src_test, which runs before src_install (see [bugs.gentoo.org/402581](https://bugs.gentoo.org/402581)). Could you possibly write and run a tiny Python program in there? That could work.
> 
> Thanks. I don't think that helps here, since I also wanted to check (for future bumps) that their tools install the py-so-lib into a correct directory. Some previous version installed the `*.so` into `$prefix/lib/python3.X/site-packages/LHAPDF-$LHAPDF_VERSION-py3.X-linux-x86_64.egg/` and now `$prefix/lib/python3.X/site-packages/`.
> 
> Just checking that the py-so-lib does not return rubbish might work, but here `cteq66` data gets installed to `/usr/share/LHAPDF/` and it is needed for testing the uncertainties. So one would have to adjust the environment `LHAPDF_DATA_DIR` and manually fetch the `cteq66` data into that directory (or create fake data). Maybe upstream can integrate something like this into their tests.

It's possible to do an emergency sanity check in src_install or even pkg_postinst, if you want.

File: sci-physics/lhapdf/lhapdf-6.5.2.ebuild [hunk 1]
```diff
@@ -32,7 +32,9 @@ RDEPEND="
 	dev-libs/boost:=
 	${PYTHON_DEPS}"
 DEPEND="${RDEPEND}"
-
+PATCHES=(
+	"${FILESDIR}"/${P}-py.patch
```
Comment (AndrewAmmerlaan) [inline sci-physics/lhapdf/lhapdf-6.5.2.ebuild:36]: This ebuild should be revbumped. Otherwise LGTM :+1:

**Comment by APN-Pucky** [comment]
Something is still wrong...
I tested it again on a different machine and I get:
```py
>>> import lhapdf
>>> pdfset = lhapdf.getPDFSet('cteq66')
>>> u = pdfset.uncertainty([0.5 for k in range(pdfset.size)])
>>> u.central
0.5
>>> u.central
4.6729839406581e-310
```

**Comment by APN-Pucky** [comment]
Seems to be a upstream issue. I get the same once working behavior from in a ubuntu docker with lhapdf-6.5.3

**Pull request closed without merging**